src/OpenIddict.Client.WebIntegration/OpenIddictClientWebIntegrationHandlers.cs

```diff
@@ -1862,6 +1862,18 @@ public ValueTask HandleAsync(ProcessChallengeContext context)
                 context.Request["resource"] = settings.Resource;
             }
 
+            // By default, Alibaba Cloud doesn't return a refresh token for native applications but allows sending an
+            // "access_type" parameter to retrieve one (but it is only returned during the first authorization dance).
+            // The documentation also indicates the "prompt" parameter is supported but not required,
+            // which can be set to "admin_consent" to force the display of the authorization page
+            if (context.Registration.ProviderType is ProviderTypes.AlibabaCloud)
+            {
+                var settings = context.Registration.GetAlibabaCloudSettings();
+
+                context.Request["access_type"] = settings.AccessType;
+                context.Request["prompt"] = settings.Prompt;
```

Comment:
```suggestion
                context.Request.Prompt = settings.Prompt;
```

You can directly the `Prompt` property since it's a standard OIDC parameter exposed by the `OpenIddictRequest` primitive 😃

Reply:
While you're changing that, let's also change line 1862 to use the same pattern:

```csharp
context.Request.Resources = [settings.Resource];
```

Reply:
Completed.

I also changed `"display"` to the `Display` property in Huawei and Weibo providers. I hope that I haven't made things worse.

Reply:
Excellent, thanks!

> I also changed `"display"` to the `Display` property in Huawei and Weibo providers. I hope that I haven't made things worse.

Great initiative 👍🏻

Reply:
```diff
- context.Request["resource"] = settings.Resource;
+ context.Request.Resources = [settings.Resource];
```

The type of `"resource"` is changed from a `string?` to a `string?[]?` likely resulting in a breaking change.

Reply:
> The type of "resource" is changed from a `string?` to a `string?[]?` likely resulting in a breaking change.

How that property is represented internally by the `OpenIddictRequest` primitive doesn't affect how it's eventually represented as an OAuth 2.0 parameter: for authorization request parameters, multi-valued parameters are always represented as multiple parameters (which was explicitly forbidden in the initial OAuth 2.0 specification, but was allowed in subsequent specifications, including the specification that defines the standard version of the `resource` parameter). Here, since there's just one value, the representation in the authorization request doesn't actually change at all.

That said, I had not realized that setting was optional in the AD provider, so it's possible that an empty parameter value may be set by using the property. Eventually, there shouldn't be any difference in the actual authorization request as OpenIddict excludes parameters with empty values when generating the URI (e.g multiple `resource` parameters with `string.Empty` or `null` as the value), but you're right it's technically a behavior change that is observable from custom event handlers, so I reverted it (I just used the appropriate constant since it's a standard parameter) 👍🏻

Reply:
When I made the change, I did have doubts about whether it was a breaking change, which is why I wrote that deleted reply. Later, however, I realized it was about the internal handling of parameters, so I deleted it.

I truly couldn't think of or understand if it would have any other impacts. I appreciate that you considered and pointed it out, which is truly admirable. It seems that raising reasonable doubts is never a bad idea after all. 👍

Reply:
> I appreciate that you considered and pointed it out, which is truly admirable. It seems that raising reasonable doubts is never a bad idea after all. 👍

🙏🏻 

(using the dedicated property actually looked nicer so I pushed a commit to bring it back but guard the call with a `string.IsNullOrEmpty()` check 😄)